src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,523 @@
+                ) {
+                    const colorHue = generateColor(
+                        dataTarget.hue.lowerShade,
+                        dataTarget.hue.upperShade,
+                        dataTarget,
+                        useYAxisData(d.weight)
+                    );
+                    return colorHue(bubbleScale(d.weight));
+                }
+                if (
+                    isHueDefined(dataTarget.hue) &&
+                    areWeightDefined(dataTarget) === false &&
+                    useYAxisData(d.weight)
+                ) {
+                    const colorHue = generateColor(
+                        dataTarget.hue.lowerShade,
+                        dataTarget.hue.upperShade,
+                        dataTarget,
+                        useYAxisData(d.weight)
+                    );
+                    return colorHue(d.y);
+                }
+                return dataTarget.color;
+            })
+            .style("opacity", constants.DEFAULT_BUBBLE_OPACITY)
+            .attr("stroke", constants.DEFAULT_BUBBLE_STROKE);
+    };
+
+    const renderSelectionPath = (path, value, index) => {
+        path.append("g")
+            .classed(styles.dataPointSelection, true)
+            .attr("transform", transformPoint(scale)(value))
+            .attr("aria-disabled", utils.isDefined(value.onClick))
+            .attr("aria-hidden", true)
+            .attr("aria-describedby", value.key)
+            .append("circle")
+            .attr("r", (d) => {
+                if (
+                    areWeightDefined(dataTarget) &&
+                    d.weight !== undefined &&
+                    !utils.isDefined(dataTarget.weight.maxRadius)
+                ) {
+                    return (
+                        bubbleScale(d.weight) +
+                        constants.DEFAULT_BUBBLE_SELECTOR_RADIUS
+                    );
+                }
+                if (utils.isUndefined(dataTarget.weight)) {
+                    return (
+                        constants.DEFAULT_BUBBLE_RADIUS_MAX +
+                        constants.DEFAULT_BUBBLE_SELECTOR_RADIUS
+                    );
+                }
+                return (
+                    dataTarget.weight.maxRadius +
+                    constants.DEFAULT_BUBBLE_SELECTOR_RADIUS
+                );
+            });
+    };
+    pointGroupPath
+        .append("g")
+        .classed(styles.pointGroup, true)
+        .each(function(d, i) {
+            const dataPointSVG = d3.select(this);
+            renderSelectionPath(dataPointSVG, d, i);
+            renderDataPoint(dataPointSVG, d, i);
+        });
+};
+
+/**
+ * Checks if the weight object is defined with min and max values.
+ *
+ * @private
+ * @param {object} dataTarget - data for the bubble graph
+ * @returns {boolean} - returns true if weight is defined and inside weight min and max is also defined else false.
+ */
+const areWeightDefined = (dataTarget) =>
```

Comment:
If you are using **are** then can you make this as `areWeightsDefined`

Reply:
Yes, I will correct it.